### PR TITLE
fix: [CDS-75443]: improve the error message for elastic group setup

### PR DESCRIPTION
**Pull request by vlica-harness**
## Describe your changes
Previously how it was the error message:
image

Improved error message to have the execution logs with the exception message and also the json.
Screenshot 2023-08-03 at 13 45 13

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51333)

### CODE REVIEW

**Comment by vlica-harness** [comment]
trigger uts_JDK11

**Comment by vlica-harness** [comment]
trigger messagecheck

File: 125-cd-nextgen/src/main/java/io/harness/cdng/elastigroup/ElastigroupSetupStep.java [hunk 1]
```diff
@@ -100,8 +112,24 @@ public TaskChainResponse executeElastigroupTask(Ambiance ambiance, StepElementPa
...
+          ElastigroupCommandUnitConstants.CREATE_ELASTIGROUP.toString(), ambiance, true);
+      logCallback.saveExecutionLog(color(format("Failed to parse the ElasticGroup Json file, cause: %s",
+                                             e.getCause() != null && StringUtils.isNotBlank(e.getCause().getMessage())
+                                                 ? e.getCause().getMessage()
+                                                 : ExceptionUtils.getMessage(e)),
+                                       LogColor.Red, LogWeight.Bold),
+          ERROR);
+      logCallback.saveExecutionLog(
+          color(format("%s", executionPassThroughData.getElastigroupConfiguration()), LogColor.White, LogWeight.Normal),
+          ERROR, CommandExecutionStatus.FAILURE);
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/elastigroup/ElastigroupSetupStep.java:130]: can we check if this problem exists with other ECS steps as well ?

File: 125-cd-nextgen/src/main/java/io/harness/cdng/elastigroup/ElastigroupSetupStep.java [hunk 2]
```diff
@@ -100,8 +112,24 @@ public TaskChainResponse executeElastigroupTask(Ambiance ambiance, StepElementPa
...
           elastigroupSetupStepParameters.getInstances(), ambiance);
     } catch (Exception e) {
+      unitProgressData.getUnitProgresses().add(
+          UnitProgress.newBuilder()
+              .setUnitName(ElastigroupCommandUnitConstants.CREATE_ELASTIGROUP.toString())
+              .setStatus(UnitStatus.FAILURE)
+              .build());
+      LogCallback logCallback = elastigroupStepCommonHelper.getLogCallback(
+          ElastigroupCommandUnitConstants.CREATE_ELASTIGROUP.toString(), ambiance, true);
+      logCallback.saveExecutionLog(color(format("Failed to parse the ElasticGroup Json file, cause: %s",
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/elastigroup/ElastigroupSetupStep.java:122]: is should be Elasti ( without the C )

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]

**Comment by vlica-harness** [comment]
trigger uts_JDK11

File: 125-cd-nextgen/src/main/java/io/harness/cdng/elastigroup/ElastigroupSetupStep.java (hunk 2 above)
Comment (vlica-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/elastigroup/ElastigroupSetupStep.java]: good catch, thanks

File: 125-cd-nextgen/src/main/java/io/harness/cdng/elastigroup/ElastigroupSetupStep.java (hunk 1 above)
Comment (vlica-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/elastigroup/ElastigroupSetupStep.java:130]: I checked for EcsRollingDeploy, EcsCanaryDeploy when json manifest is malrformed, and the error is pretty meaningful. For example:
ecs rolling deser

**Review by arvind-choudhary-h — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51333)* status: 2 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51333)* status: :shipit: complete! ✓

**Comment by yogesh-chauhan** [comment]
trigger ut4

**Comment by yogesh-chauhan** [comment]
trigger review

**Pull request merged**